### PR TITLE
TNO-1501: View evening overview on MMI Subscriber

### PR DESCRIPTION
**Pull request by jdtoombs**
In this PR:

- new sidebar item to view evening overview
- click to view today's overview
- click logo in top left to navigate home

image

### CODE REVIEW

File: app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        );
+        return response.data;
+      },
+      previewAVOverview: async (instanceId: number) => {
+        const response = await dispatch<IReportResultModel>('preview-av-overview', () =>
+          api.previewAVOverview(instanceId),
+        );
+        return response.data;
+      },
+      publishAVOverview: async (instanceId: number) => {
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts:59]: Subscribers cannot publish the Evening Overview

File: app/subscriber/src/features/daily-overview/AVOverviewPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [, setInstance] = React.useState<IAVOverviewInstanceModel>();
+  const [preview, setPreview] = React.useState<IReportResultModel | undefined>();
+  const [isPublished, setIsPublished] = React.useState(false);
+
+  const handlePreviewReport = React.useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const instance = await findAVOverview(new Date().toISOString());
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/AVOverviewPreview.tsx:18]: Are they not allowed to view prior days too?

File: app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return response.data;
+      },
+      getAVOverview: async (id: number) => {
+        const response = await dispatch<IAVOverviewInstanceModel | undefined>(
+          'get-av-overview',
+          () => api.getAVOverview(id),
+        );
+        return response.data;
+      },
+      addAVOverview: async (model: IAVOverviewInstanceModel) => {
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts:35]: Subscribers are not allowed to do this

File: app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        const response = await dispatch<IAVOverviewInstanceModel>('add-av-overview', () =>
+          api.addAVOverview(model),
+        );
+        return response.data;
+      },
+      updateAVOverview: async (model: IAVOverviewInstanceModel) => {
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts:41]: Subscribers are not allowed to do this

File: app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        const response = await dispatch<IAVOverviewInstanceModel>('update-av-overview', () =>
+          api.updateAVOverview(model),
+        );
+        return response.data;
+      },
+      deleteAVOverview: async (model: IAVOverviewInstanceModel) => {
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts:47]: Subscribers are not allowed to do this

File: app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  const controller = React.useMemo(
+    () => ({
+      findAVOverview: async (publishedOn: Date | string) => {
+        const response = await dispatch<IAVOverviewInstanceModel | undefined>(
+          'find-av-overviews',
+          () => api.findAVOverview(publishedOn),
+        );
+        return response.data;
+      },
+      getAVOverview: async (id: number) => {
```
Comment (Fosol) [inline app/subscriber/src/features/daily-overview/hooks/useAVOverviewInstances.ts:28]: I don't think subscribers need this either.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/subscriber/src/features/daily-overview/AVOverviewPreview.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/subscriber/src/features/daily-overview/AVOverviewPreview.tsx:18]: At the moment I just based it off the UI, provided but I can create another ticket to cover this!

**Review by Fosol — APPROVED** [review]

**Pull request merged**